clang/lib/Frontend/CompilerInvocation.cpp

```diff
@@ -3618,25 +3618,6 @@ bool CompilerInvocation::ParseLangArgs(LangOptions &Opts, ArgList &Args,
       LangStd = OpenCLLangStd;
   }
 
-  if (Args.hasArg(OPT_fsycl_is_device) || Args.hasArg(OPT_fsycl_is_host)) {
```

Comment:
Note: this code was entirely dead - the tablegenerated option parsing code just below will reset `SYCLVersion` to whatever was parsed on the cc1 command line or `SYCL_None`. Removing the code changed zero tests, so I assume none of this was actually necessary.

Reply:
Did you check what happens if an invalid value is passed? Looks like the code below intended  to diagnose that.

Reply:
We appear to have no existing test coverage for that, neat! :-D I'll add some for garbage values, but the `"1.2.1", "121", "sycl-1.2.1"` stuff is already handled automatically by the option parsing stuff, so I don't expect any code changes (just additional test coverage).

Reply:
Test coverage added, and I learned something new today. You can't put multiple cc1 command line arg tests into a single `TEST_F` fixture because the test harness will exhibit truly *bizarre* behavior (arguments in the generated would get overwritten when you sneezed near them). Hence why it's multiple tests rather than one test.

Reply:
> Test coverage added, and I learned something new today. You can't put multiple cc1 command line arg tests into a single `TEST_F` fixture because the test harness will exhibit truly _bizarre_ behavior (arguments in the generated would get overwritten when you sneezed near them). Hence why it's multiple tests rather than one test.

Thanks @AaronBallman for sharing the information.